pkg/cosign/kubernetes/secret.go

```diff
@@ -19,16 +19,39 @@ import (
 	"fmt"
 	"io/ioutil"
 	"os"
+	"regexp"
 	"strings"
 
 	"github.com/pkg/errors"
+	"github.com/sigstore/cosign/pkg/cosign"
 	v1 "k8s.io/api/core/v1"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
-	"github.com/sigstore/cosign/pkg/cosign"
 )
 
+var referenceRegex = regexp.MustCompile(`^k8s:\/\/[a-zA-Z0-9\-]+(\/{1}[a-zA-Z0-9\-]+)?$`)
```

Comment:
I prefer to avoid regexes for stuff like this if possible... Is there anyway to parse this like a uri? or strip the prefix and then split on a /?

Reply:
you are absolutely right, thank you so much for this suggestion, I updated the code according to your feedback.